### PR TITLE
Adding minor tests to test_regtap

### LINKED ISSUE #425: MAINT: improve test coverage of regtap
Currently, it's only 55% with missing coverage of numerous methods leading to wonder whether hash-based splitting of standardid works as intended for services with IDs including a hash, e.g. the distinction of SIA and SIA2.

https://github.com/astropy/pyvo/blob/main/pyvo/registry/regtap.py#L311
https://github.com/astropy/pyvo/blob/main/pyvo/registry/regtap.py#L338

### PR DESCRIPTION
**Pull request by msdemlei**
(in an attempt to tackle #425).

closes #425

### CODE REVIEW

**Review by bsipocz — APPROVED** [review]
Thank you for looking into this!

File: pyvo/registry/regtap.py [hunk 1]
```diff
@@ -539,7 +539,11 @@ def region_of_regard(self):
         by which a positional query against this resource should be "blurred"
         in order to get an appropriate match.
         """
-        return float(self.get("region_of_regard", 0))
+        # we get NULLs as NaNs here
+        val = self["region_of_regard"]
+        if val != val:
```
Comment (bsipocz) [inline pyvo/registry/regtap.py:544]: I'm not sure this is the best way to test for `np.nan`s, or what other values would you expect here?

(e.g. `np.isnan(val)` or `np.isfinite(val)` is a cleaner check for NaNs)

**Review by bsipocz — CHANGES_REQUESTED** [review]
Actually, there is one point where a more pythonic testing could be used,  but I'm not fully aware of what actual input values we expect.

**Comment by msdemlei** [comment]
On Wed, Apr 26, 2023 at 11:13:45AM -0700, Brigitta Sipőcz wrote:
> @bsipocz requested changes on this pull request.
> I'm not sure this is the best way to test for `np.nan`s, or what
> other values would you expect here?

Yeah, it's (possibly) NaNs (the universal null value for floats in
VOTables, whether or not we like that).  I had wanted to avoid
another import, and x!=x is a rather safe test for "you don't want to
confront normal user code with this" -- but yes, the intent is
clearer with numpy.isnan, so I've force-pushed a change to that
effect.

**Comment by bsipocz** [comment]
> I had wanted to avoid another import

numpy is practically imported at that point with astropy

**Review by bsipocz — APPROVED** [review]

**Pull request merged**